src/base/static/js/views/layer-view.js

```diff
@@ -42,22 +77,32 @@
 
       // Don't draw new places. They are shown by the centerpoint in the app view
       if (!this.model.isNew() && this.isPublishable()) {
+        var geom,
+            styleRule = {},
+            zoomRule = {},
+            styleRuleContext = _.extend({},
+              this.model.toJSON(),
+              {map: {zoom: this.map.getZoom()}},
+              {layer: {focused: this.isFocused}});
+
+        for (var i = 0; i < this.styleRules.length; i++) {
+          if (this.styleRules[i].apply(styleRuleContext)) {
+            styleRule = this.styleRules[i].props;
+            break;
+          }
+        }
 
-        // Determine the style rule to use based on the model data and the map
-        // state.
-        context = _.extend({},
-          this.model.toJSON(),
-          {map: {zoom: this.map.getZoom()}},
-          {layer: {focused: this.isFocused}});
-        // Set the icon here:
-        this.styleRule = L.Argo.getStyleRule(context, this.placeType.rules);
-
-        // Zoom checks here, for overriding the icon size, anchor, and focus icon:
-        if (this.placeType.hasOwnProperty('zoomType')) {
-          var zoomRules = this.options.placeTypes[this.placeType.zoomType];
-          this.styleRule = L.Argo.getZoomRule(this.styleRule, zoomRules);
+        for (var i = 0; i < this.zoomRules.length; i++) {
```

Comment:
Re-add the comment here: `// Zoom checks here, for overriding the icon size, anchor, and focus icon:`